src/Castle.Windsor/MicroKernel/Arguments.cs

```diff
@@ -110,13 +92,17 @@ bool IDictionary.IsReadOnly
 			get { return arguments.IsReadOnly; }
```

Comment:
`Arguments.IsReadOnly` should return `isReadOnly` not the internal dictionary's state.